docs/man.jpconj/man3/Export.3
.TH "Export" 3 "Tue Aug 29 2017" "Version 2.0.0" "JapKatsuyou.JpConj" \" -*- nroff -*-
.ad l
.nh
.SH NAME
Export \- 
.SH SYNOPSIS
.br
.PP
.PP
\fC#include <export\&.h>\fP
.SS "Public Member Functions"

.in +1c
.ti -1c
.RI "\fBExport\fP ()"
.br
.ti -1c
.RI "void \fBaddContent\fP (QString content)"
.br
.ti -1c
.RI "void \fBsetStyle\fP (QString styleFile)"
.br
.ti -1c
.RI "void \fBaddStyle\fP (QString styleContent)"
.br
.ti -1c
.RI "void \fBsetRTL\fP (bool RTL)"
.br
.ti -1c
.RI "void \fBexportPdf\fP (QString filename)"
.br
.ti -1c
.RI "void \fBexportOdf\fP (QString filename)"
.br
.ti -1c
.RI "void \fBexportHtml\fP (QString filename)"
.br
.ti -1c
.RI "void \fBprint\fP (QPrinter *printer)"
.br
.ti -1c
.RI "void \fBresetContent\fP ()"
.br
.in -1c
.SS "Static Public Member Functions"

.in +1c
.ti -1c
.RI "static void \fBsetConfigExportParts\fP (QMap< QString, bool > exportParts)"
.br
.ti -1c
.RI "static bool \fBgetConfigExportPart\fP (QString exportPart)"
.br
.in -1c
.SH "Constructor & Destructor Documentation"
.PP 
.SS "Export::Export ()"

.SH "Member Function Documentation"
.PP 
.SS "void Export::addContent (QString content)"

.SS "void Export::addStyle (QString styleContent)"

.SS "void Export::exportHtml (QString filename)"

.SS "void Export::exportOdf (QString filename)"

.SS "void Export::exportPdf (QString filename)"

.SS "bool Export::getConfigExportPart (QString exportPart)\fC [static]\fP"

.SS "void Export::print (QPrinter * printer)"

.SS "void Export::resetContent ()"

.SS "void Export::setConfigExportParts (QMap< QString, bool > exportParts)\fC [static]\fP"

.SS "void Export::setRTL (bool RTL)"

.SS "void Export::setStyle (QString styleFile)"


.SH "Author"
.PP 
Generated automatically by Doxygen for JapKatsuyou\&.JpConj from the source code\&.


CHANGELOG.md

# json-VCard Changelog

This is the changelog syntax:
* NEW: For any new module or file (creation)
* ADD: For any functions added inside a module (addition)
* IMPROVE: For any improvements to existing materials.
This affects the behavior, but doesn't break any precedent
* MODIFY: Modification of function names, files names and directories.
This will break the projects developed on precedent versions.
* FIX: Fixing bugs
* DEL: Delete or deprecate some functions

## Version 2.0.0 (2016-07-03)

* ADD: Control the font (size and type) for both default language and Japanese
* NEW: Choose the icons set
* NEW: Show examples in default language
* ADD: Show Furigana and Romaji pronunciation of the verb

## Version 1.0.1 (2014-01-01)

* I don't know what changed :)

## Version 0.2.0 (2013-09-04)

* NEW: first release


jpconj/UiHelper/export.cpp
/*
    This file is part of JapKatsuyou project; an application that provide
    Japanese verb conjugation

    Copyright (C) 2013-2016  AnqaFalak group
    Copyright (C) 2013  Abdelkrime Aries <kariminfo0@gmail.com>

    Authors:
            Abdelkrime Aries <kariminfo0@gmail.com>

  Licensed under the Apache License, Version 2.0 (the "License");
  you may not use this file except in compliance with the License.
  You may obtain a copy of the License at

      http://www.apache.org/licenses/LICENSE-2.0

  Unless required by applicable law or agreed to in writing, software
  distributed under the License is distributed on an "AS IS" BASIS,
  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  See the License for the specific language governing permissions and
  limitations under the License.
 */

#include "export.h"


/*******************************************************
 *                    PUBLIC
 *******************************************************/


Export::Export()
{
    resetContent();
    isRTL = false;
}

void Export::addContent(QString content)
{
    htmlContent.replace("<!--CONTENT-->",
                        content + "\n<!--CONTENT-->");
}

void Export::addStyle(QString styleContent)
{
    QString CSS = "<style type=\"text/css\">\n";
    CSS += styleContent + "\n";
    CSS += "</style>\n";
    htmlContent.replace("<!--STYLESHEETS-->", CSS + "\n<!--STYLESHEETS-->\n");
}

void Export::setStyle(QString styleFile)
{
    this->styleFile = styleFile;
}

void Export::setRTL(bool RTL)
{
    isRTL = RTL;
    if (RTL)
        htmlContent.replace("<body><!--DIR-->", "<body dir=\"rtl\">");
    else
        htmlContent.replace("<body><!--DIR-->", "<body dir=\"ltr\">");
}

void Export::exportPdf(QString filename)
{
    QPrinter printer(QPrinter::HighResolution);
    printer.setOutputFormat(QPrinter::PdfFormat);
    printer.setPaperSize(QPrinter::A4);
    printer.setOutputFileName(filename);

    QWebView webview;
    webview.setHtml(htmlContent);


    if(styleFile.length()>0){
        QWebSettings * settings = webview.settings();
        settings->setUserStyleSheetUrl(QUrl("file://" + styleFile));
    }

    webview.print(&printer);
}

void Export::exportOdf(QString filename)
{
    QTextDocument * doc = new QTextDocument();
    doc->setHtml(htmlContent);

    //reading style file
    /*QFile cssFile(styleFile);
    QString styleSheet = "";
    if (cssFile.open(QFile::ReadOnly | QFile::Text)){
        QTextStream cssStream(&cssFile);
        styleSheet = cssStream.readAll();
        cssFile.close();
    }
    //end reading
    doc->setDefaultStyleSheet(styleSheet);*/
    QTextDocumentWriter writer(filename);
    writer.setFormat("odf");
    writer.write(doc);
}


void Export::exportHtml(QString filename)
{
    QFile outputFile(filename);
    if(!outputFile.open(QIODevice::WriteOnly)){
        qDebug() << "- Error, unable to open the file for output";
        return;
    }

    QTextStream outStream(&outputFile);
    outStream.setCodec("UTF-8");

    if(styleFile.length()>0){
        QFileInfo styleInfo(styleFile);
        QString stylename = styleInfo.fileName();

        QFileInfo destInfo(filename);
        QString stylefolder = destInfo.fileName() + "_files";
        QString styleto = destInfo.absolutePath() + "/" + stylefolder + "/" + stylename;

        QDir(destInfo.absolutePath()).mkdir(stylefolder);
        qDebug() << styleFile << " to " << styleto;
        QFile::copy(styleFile, styleto);

        addStyle("./" + stylefolder + "/" + stylename);
    }

    outStream << htmlContent;

    outputFile.close();
}



void Export::print(QPrinter *printer)
{
    QWebView webview;
    webview.setHtml(htmlContent);

    if(styleFile.length()>0){
        QWebSettings * settings = webview.settings();
        settings->setUserStyleSheetUrl(QUrl("file://" + styleFile));
    }

    webview.print(printer);
}



void Export::resetContent()
{
    htmlContent = "";
    htmlContent += "<html>\n";
    htmlContent += "<head>\n";
    htmlContent += "<title>JapKatsuyou</title>\n";
    htmlContent += "<meta charset=\"UTF-8\">\n"; //<meta http-equiv=\"Content-Type\" content=\"text/html; charset=utf-8\">\n";
    htmlContent += "<!--STYLESHEETS-->\n";
    htmlContent += "<body><!--DIR-->\n";
    htmlContent += "<!--CONTENT-->\n";
    htmlContent += "</body>\n";
    htmlContent += "</html>\n";
}



void Export::setConfigExportParts(QMap<QString, bool> exportParts)
{
    QSettings settings;//("DzCoding", "JapKatsuyou")
    settings.beginGroup("Export");
    foreach (QString part, exportParts.keys())
        settings.setValue(part, exportParts.value(part));
    settings.endGroup();

}

bool Export::getConfigExportPart(QString exportPart)
{
    bool result;
    QSettings settings;//("DzCoding", "JapKatsuyou")
    settings.beginGroup("Export");
    result = settings.value(exportPart, true).toBool();
    settings.endGroup();
    //qDebug() << exportPart << " = " << result;
    return result;
}
